pkg/arvo/sys/hoon.hoon

```diff
@@ -6244,6 +6244,7 @@
   ~/  %scow
   |=(mol/dime ~(rend co %$ mol))
 ++  slat  |=(mod/@tas |=(txt/@ta (slaw mod txt)))
+++  slan  |=(mod/@tas |=(txt/@ta (need (slaw mod txt))))
```

Comment:
Is this intended for use with `+se:dejs:format`?

Reply:
No, this is intended for use with this sort of formatter. I used this 3 times, so I figured put it in zuse.

https://github.com/urbit/urbit/blob/ff7b2ab7d4f8e4c8daf28b474482b6874301effd/pkg/arvo/lib/chat-json.hoon#L14

Reply:
https://github.com/urbit/urbit/blob/ff7b2ab7d4f8e4c8daf28b474482b6874301effd/pkg/arvo/lib/invite-json.hoon#L9

Reply:
These functions don't look wrong, but +slan, +slav, and +slat all look like cruft to me.  +slaw is the only function that has any meat here; the others either curry it, wrap it in an assertion, or both.

I feel like we'd be better served by a combinator that wraps a function in a +need call.  Maybe +slaw should also be defined curried, since uncurrying is (at least a little) more convenient than currying.  That way if you're composing these functions into your own combinators, you can stay functional, but we won't need so many functions that are almost exactly the same.

Reply:
Something like this (unless we "bake" it):
```
++  want
  |*  f=$-(* (unit))
  |*  x=*
  (need (f x))
```

Reply:
Or, you know, fix currying and function composition.

Reply:
@loganallenc, is this use case still relevant given the now-general form of `+se:dejs:format`?

Reply:
@joemfb no longer relevant. Have removed it from my most recent push.

Should be ready to accept or review again at this point, addressed all listed concerns.

@belisarius222

Reply:
> Or, you know, fix currying and function composition.

I suspect the trickiest part would be letting wet gates nest inside dry gates, or equivalent functionality.  Makes think fully-baked Hoon might be a good idea ...